lib/datadog/core/configuration/option.rb

```diff
@@ -102,6 +191,29 @@ def context_eval(&block)
           @context.instance_eval(&block)
         end
 
+        def set_value_from_env_or_default
+          value = nil
+          precedence = nil
+
+          if definition.env && ENV[definition.env]
+            value = coerce_env_variable(ENV[definition.env])
+            precedence = Precedence::PROGRAMMATIC
+          end
+
+          if value.nil? && definition.deprecated_env && ENV[definition.deprecated_env]
+            value = coerce_env_variable(ENV[definition.deprecated_env])
+            precedence = Precedence::PROGRAMMATIC
```

Comment:
That comment should be changed. 

If a customer has modified the configuration value via ENV variable, it has been configured programmatically since the customer is not using the default value. 

For example: 

```ruby
option :enabled do |o|
  o.type :bool
  o.env 'DD_APPSEC_ENABLED'
  o.default false
end
```

If the customer changes the value using `Datadog.configuration.appsec.enabled = true` or via `DD_APPSEC_ENABLED`, we would want the helper function `using_default?` to return false `Datadog.configuration.appsec.using_default?(:enabled) #=> false`, since the intent of `DEFAULT = [0, :default].freeze` is to signal the customer hasn't set the value in anyway

Reply:
I see. Seems reasonable -- either updating the comment or perhaps introducing a new environment variable level seems OK.